src/components/signin/components/Signin.js

```diff
@@ -35,16 +36,34 @@ class Signin extends React.Component {
                 password,
             };
             signinRequest(payload);
-            this.setState({ visible: false });
         });
-
     }
 
     saveFormRef = formRef => {
         this.formRef = formRef;
     }
 
+    componentDidUpdate(){
+        const { 
+            isSuccessful, 
+            actions: { resetSignState }, 
+            signinfailMessage, 
+            signinStatus,
+        } = this.props;
+        const { visible } = this.state;
+    
+        if (visible && isSuccessful) {
+            this.setState({ visible: false });
+            resetSignState();
+        }
+        if ( signinStatus ) {
```

Comment:
if `signinError` (or something like this) not `signinStatus`. This is confusing. It reads if there is ever a signin status show an error message.